### PR TITLE
Add support for Sidekiq 7

### PR DESCRIPTION
**Pull request by dwightwatson**
In response to issue #50 and reading through the Sidekiq source I believe this should fix compatibility with Sidekiq 7.

### CODE REVIEW

File: lib/active_job/uniqueness/sidekiq_patch.rb [hunk 1]
```diff
@@ -91,6 +91,10 @@ def delete_by_value(name, value)
 
 # Global death handlers are introduced in Sidekiq 5.1
 # https://github.com/mperham/sidekiq/blob/e7acb124fbeb0bece0a7c3d657c39a9cc18d72c6/Changes.md#510
-if sidekiq_version >= Gem::Version.new('5.1')
+if sidekiq_version >= Gem::Version.new('7.0')
+  Sidekiq.configure_server do |config|
+    config.death_handlers << ->(job, ex) { ActiveJob::Uniqueness.unlock_sidekiq_job!(job) }
```
Comment (sharshenov) [inline lib/active_job/uniqueness/sidekiq_patch.rb:96]:
```suggestion
    config.death_handlers << ->(job, _ex) { ActiveJob::Uniqueness.unlock_sidekiq_job!(job) }
```
I guess Rubocop will complain about the variable name if it is not bein used in the block

**Review by sharshenov — CHANGES_REQUESTED** [review]
Thank you for the PR :+1: 

Could you also add a gemfile for Sidekiq 7 via [Appraisals](https://github.com/veeqo/activejob-uniqueness/blob/ae920feeb76cb82ee6e181e3e08bd95c31f1779f/Appraisals) and amend [the build matrix](https://github.com/veeqo/activejob-uniqueness/blob/ae920feeb76cb82ee6e181e3e08bd95c31f1779f/.github/workflows/main.yml) to run specs against Sidekiq 7 as well?

File: lib/active_job/uniqueness/sidekiq_patch.rb (hunk 1 above)
Comment (sharshenov) [reply lib/active_job/uniqueness/sidekiq_patch.rb:96]: Yep, It [does complain](https://github.com/veeqo/activejob-uniqueness/actions/runs/3360350642/jobs/5604585189#step:5:22)

Comment (dwightwatson) [reply lib/active_job/uniqueness/sidekiq_patch.rb]: Whoops - don't know how I stuffed that considering it's the same as the line below it.

**Comment by dwightwatson** [comment]
Made the changes - but looks like you'll need to trigger a re-run to see if they work as expected.

**Comment by sharshenov** [comment]
Thanks! It appears the `gemfiles/sidekiq_7.x.gemfile` [is not committed](https://github.com/veeqo/activejob-uniqueness/actions/runs/3376719141/jobs/5639187427#step:4:11). Have you run `appraisal install`? [link](https://github.com/thoughtbot/appraisal#usage)

**Comment by dwightwatson** [comment]
No - I've never used appraisal before so I'm not familiar with that step - sorry. Just ran it now and committed the file.

**Comment by olimart** [comment]
👋 hello, is something missing here?

**Comment by adis-io** [comment]
> 👋 hello, is something missing here?

Tests are [failed](https://github.com/veeqo/activejob-uniqueness/actions/runs/3392947985/jobs/5644699406)

**Comment by dwightwatson** [comment]
Sorry - I didn't see the failure. The tests don't run until they are triggered so I don't see the output.

Curiously the test is passing for me locally. Would be open to any insight as to what is going wrong.

**Comment by adis-io** [comment]
> Sorry - I didn't see the failure. The tests don't run until they are triggered so I don't see the output.
> 
> Curiously the test is passing for me locally. Would be open to any insight as to what is going wrong.

Are you sure you are using correct bundle?
```sh
BUNDLE_GEMFILE=gemfiles/sidekiq_7.x.gemfile bundle
BUNDLE_GEMFILE=gemfiles/sidekiq_7.x.gemfile rspec './spec/active_job/uniqueness/sidekiq_patch_spec.rb[1:7:1:1:1]'

Run options: include {:ids=>{"./spec/active_job/uniqueness/sidekiq_patch_spec.rb"=>["1:7:1:1:1"]}}
2022-11-16T21:55:19.318Z pid=98327 tid=27yj INFO: Sidekiq 7.0.1 connecting to Redis with options {:size=>5, :pool_name=>"internal", :url=>nil}
F

Failures:

  1) Sidekiq patch job death when queue adapter is Sidekiq when job class has unique strategy enabled releases the lock
     Got 0 failures and 2 other errors:
     Shared Example Group: "locks release" called from ./spec/active_job/uniqueness/sidekiq_patch_spec.rb:129

     1.1) Failure/Error: job = ActiveJob::Base.deserialize(job_data.fetch('args').first)

          NameError:
            uninitialized constant MyJob
          # ./lib/active_job/uniqueness/sidekiq_patch.rb:11:in `unlock_sidekiq_job!'
          # ./lib/active_job/uniqueness/sidekiq_patch.rb:51:in `delete'
          # ./lib/active_job/uniqueness/sidekiq_patch.rb:58:in `clear'
          # ./spec/active_job/uniqueness/sidekiq_patch_spec.rb:124:in `block (3 levels) in <top (required)>'
          # ./spec/support/active_job_helpers.rb:32:in `block (2 levels) in <top (required)>'

     1.2) Failure/Error: job = ActiveJob::Base.deserialize(job_data.fetch('args').first)

          NameError:
            uninitialized constant Analytics
          # ./lib/active_job/uniqueness/sidekiq_patch.rb:11:in `unlock_sidekiq_job!'
          # ./lib/active_job/uniqueness/sidekiq_patch.rb:51:in `delete'
          # ./lib/active_job/uniqueness/sidekiq_patch.rb:58:in `clear'
          # ./spec/active_job/uniqueness/sidekiq_patch_spec.rb:7:in `each'
          # ./spec/active_job/uniqueness/sidekiq_patch_spec.rb:7:in `block (3 levels) in <top (required)>'
          # ./spec/support/active_job_helpers.rb:32:in `block (2 levels) in <top (required)>'

Finished in 0.014 seconds (files took 0.41137 seconds to load)
1 example, 1 failure

Failed examples:

rspec './spec/active_job/uniqueness/sidekiq_patch_spec.rb[1:7:1:1:1]' # Sidekiq patch job death when queue adapter is Sidekiq when job class has unique strategy enabled releases the lock
```

**Comment by dwightwatson** [comment]
Cool - would you mind retrying the build now?

**Comment by adis-io** [comment]
> Cool - would you mind retrying the build now?

Have you run locally? I've run locally and it still fails, but anyway started tests.

**Comment by dwightwatson** [comment]
Yeah I had it green locally before pushing

**Review by sharshenov — APPROVED** [review]

**Comment by sharshenov** [comment]
Thank you!

**Pull request merged**